.travis.yml

```diff
@@ -6,23 +6,51 @@ before_install:
   - unset _JAVA_OPTIONS
 env:
   global:
-    - GRADLE_OPTS="-Xmx256m"
+    - GRADLE_OPTS='-Xmx256m'
+    - BUILD_CACHE_DIR="$HOME/build-cache"
+    - TESTS_FILE_NAME="$TRAVIS_BUILD_DIR/test_file.txt"
+    - TESTS_FILES_PREFIX='tests'
 install:
   - true
-matrix:
-  include:
-    - name: "Build core project"
-      env: BUILD_PARAMETERS='core:reportScoverage'
-    - name: "Build non core projects"
-      env: BUILD_PARAMETERS='reportScoverage -x core:reportScoverage'
-script:
-  - ./gradlew --no-daemon $BUILD_PARAMETERS
 after_success:
   - bash <(curl -s https://codecov.io/bash)
 before_cache:
   - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
   - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+  - mkdir -p $HOME/build-cache
 cache:
   directories:
     - $HOME/.gradle/caches/
     - $HOME/.gradle/wrapper/
+    - $HOME/build-cache
+jobs:
+  include:
+    - stage: 'clone repo and compile project'
+      name: 'Clone repo & compile project'
```

Comment:
same here - the name is confusing since nothing is compiled here.  Let's have is as `Clone repo & resolve dependencies`